zebra-consensus/src/primitives/groth16.rs

```diff
@@ -9,120 +9,117 @@ use std::{
 };
 
 use bellman::{
-    groth16::{PreparedVerifyingKey, Proof},
+    groth16::{batch, prepare_verifying_key, VerifyingKey},
     VerificationError,
 };
 use bls12_381::Bls12;
-use pairing::MultiMillerLoop;
-use rand::{thread_rng, CryptoRng, RngCore};
+use futures::future::{ready, Ready};
+use once_cell::sync::Lazy;
+use rand::thread_rng;
 use tokio::sync::broadcast::{channel, error::RecvError, Sender};
-use tower::Service;
-use tower_batch::BatchControl;
+use tower::{util::ServiceFn, Service};
+use tower_batch::{Batch, BatchControl};
```

Comment:
Just a reminder that #1805 wasn't actually fixed, so if you see weird hangs during batching, we should go fix it.

Maybe we should schedule it for the next sprint, since it's one of those risky bugs to leave unfixed.
cc @mpguerra.

Reply:
Ha! It is already tentatively scheduled in the next sprint.